### PR TITLE
cisco_meraki: improve event handling

### LINKED ISSUE #9700: [Cisco Meraki Events] Parse AnyConnect, splash page, martian_vlan events
Cisco Meraki MX firewalls emit a large number of event types, where only a few of them are parsed by the integration. 

Firstly, I suggest that whenever the integration does not recognize the sub-type for an event, it emits the unparsed message text (i.e everything after the `events` keyword) in a suitable field (`message`?) . Currently we have to enable the option to keep `event.original` which should not be needed. 

Secondly, here are some events related to AnyConnect VPN, Splash pages and martian vlans that should be parsed properly for the integration to be really usable. I've edited out remote IP addresses and username 

```
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[369] Peer IP=1.1.1.1 User[user.name1]: Session disconnected. Session Type: SSL, Duration: 1d:24h:00m:00s, Bytes xmt: 22412809, Bytes rcv: 4138099, Reason: Max time exceeded '
events anyconnect_vpn_disconnect user id 'user.name1' local ip 172.25.22.82 connected from 1.1.1.1
 
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[396] Peer IP=2.2.2.2 User[user.name2]: Deleted TLS tunnel[396.17] from DB. Reason: DPD kill '
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[396] Peer IP=2.2.2.2 User[user.name2]: Deleted DTLS tunnel[396.18] from DB. Reason: DPD kill '
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[396] Peer IP=2.2.2.2 User[user.name2]: conn_id[18007] Added DTLS tunnel[396.18] to DB '
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[396] Peer IP=2.2.2.2 User[user.name2]: conn_id[18006] Added TLS tunnel[396.17] to DB '
events anyconnect_vpn_connect user id 'user.name2' local ip 172.25.22.244 reconnected from 2.2.2.2
events type=anyconnect_vpn_session_manager msg= 'Peer IP=2.2.2.2 User[user.name2] Sess-ID[396]: Applied VPN filter[AnyConnect policy] for assigned IP 172.25.22.244 '
 
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[420] Peer IP=3.3.3.3 User[user.name3]: Session disconnected. Session Type: SSL, Duration: 0d:00h:23m:03s, Bytes xmt: 135325, Bytes rcv: 74821, Reason: User Requested '
events type=anyconnect_vpn_session_manager msg= 'Sess-ID[420] Peer IP=3.3.3.3 User[user3.name]: Deleted TLS tunnel[420.3] from DB. Reason: User Requested '
 
events type=anyconnect_vpn_auth_success msg= 'RADIUS[511] Server IP=172.25.30.12 Server port=1812 Peer IP=4.4.4.4 Peer port=56193 User=user.name4: Authentication request accepted. '
 
events type=splash_auth mac='4C:03:4F:5C:5B:43' duration='604800' vap='0' wired_vlan='-1' download='1Gbps' upload='1Gbps'
 
events type=martian_vlan Client='172.25.25.74' MAC='24:5E:BE:19:45:2E' VLAN='24' details='sent 7088457 unexpected packets (Last seen packet IP=169.254.100.100)'
events type=martian_vlan Client='172.25.25.196' MAC='CC:96:E5:9F:09:89' VLAN='24' details='sent 834351 unexpected packets (Last seen packet IP=172.25.40.16, IP on VLAN=40)'
events type=martian_vlan summary='8058083 unexpected packets seen (8057025 packets till last report)'
```

As a reminder, cisco documentation on event log format can be found [here](https://documentation.meraki.com/General_Administration/Monitoring_and_Reporting/Syslog_Server_Overview_and_Configuration) and [here](https://documentation.meraki.com/General_Administration/Monitoring_and_Reporting/Syslog_Event_Types_and_Log_Samples). Unfortunately this list is incomplete.

### PR DESCRIPTION
**Pull request by efd6**
## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #9700

### CODE REVIEW

File: packages/cisco_meraki/data_stream/log/elasticsearch/ingest_pipeline/events.yml [hunk 1]
```diff
@@ -270,6 +290,129 @@ processors:
...
+####################################################
+- grok:
+    field: message
+    patterns: 
+      - "msg= ?'%{NOTCOLON:_temp.left}: %{DATA:_temp.right}%{REASON}? ?'"
+    pattern_definitions:
+      NOTCOLON: '[^:]*'
+      REASON: ' Reason: %{DATA:cisco_meraki.anyconnect_vpn_session_manager.reason}'
+    if: ctx.cisco_meraki?.event_subtype == "anyconnect_vpn_session_manager"
+    # ignore_failure: true
```
Comment (andrewkroh) [inline packages/cisco_meraki/data_stream/log/elasticsearch/ingest_pipeline/events.yml:304]: Do you want ignore_failure turned on here?

File: packages/cisco_meraki/data_stream/log/elasticsearch/ingest_pipeline/events.yml [hunk 2]
```diff
@@ -270,6 +290,129 @@ processors:
...
+      DELETED: 'Deleted'
+      APPLY: 'Applied VPN %{FILTER} for assigned IP %{IP:cisco_meraki.anyconnect_vpn_session_manager.ip}'
+      SESSEND: 'Session %{DISCONNECTED:cisco_meraki.anyconnect_vpn_session_manager.action}\. Session Type: %{WORD:cisco_meraki.anyconnect_vpn_session_manager.session_type}, Duration: %{NOTSPACE:cisco_meraki.anyconnect_vpn_session_manager.duration}, Bytes xmt: %{NUMBER:cisco_meraki.anyconnect_vpn_session_manager.bytes_out}, Bytes rcv: %{NUMBER:cisco_meraki.anyconnect_vpn_session_manager.bytes_in},?'
+      DISCONNECTED: 'disconnected'
+      FILTER: 'filter\[%{NOTBRACKET:cisco_meraki.anyconnect_vpn_session_manager.filter}\]'
+      CONNID: 'conn_id\[%{NOTBRACKET:cisco_meraki.anyconnect_vpn_session_manager.conn_id}\]'
+      TUNNEL: '%{WORD:cisco_meraki.anyconnect_vpn_session_manager.tunnel_type} tunnel\[%{NOTBRACKET:cisco_meraki.anyconnect_vpn_session_manager.tunnel_id}\]'
+      NOTBRACKET: '[^\]]*'
+    if: ctx._temp?.right != null
+    # ignore_failure: true
```
Comment (andrewkroh) [inline packages/cisco_meraki/data_stream/log/elasticsearch/ingest_pipeline/events.yml:346]: Same question.

**Review by andrewkroh — APPROVED** [review]
Other than the questions, LGTM

File: packages/cisco_meraki/data_stream/log/elasticsearch/ingest_pipeline/events.yml (hunk 1 above)
Comment (efd6) [reply packages/cisco_meraki/data_stream/log/elasticsearch/ingest_pipeline/events.yml:304]: Yeah, I do. I was turning it off to find issues.

**Pull request merged**